libs/cart/src/drivers/magento/cart-shipping-methods.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { Apollo } from 'apollo-angular';
+
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { DaffCartShippingMethodsServiceInterface } from '../interfaces/cart-shipping-methods-service.interface';
+import { listShippingMethods } from './queries';
+import { DaffCartShippingRate } from '../../models/cart-shipping-rate';
+import { ListShippingMethodsResponse } from './models/responses/list-shipping-methods';
+import { DaffMagentoCartShippingRateTransformer } from './transforms/outputs/cart-shipping-rate.service';
+
+/**
+ * A service for making Magento GraphQL queries for carts' shipping methods.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffMagentoCartShippingMethodsService implements DaffCartShippingMethodsServiceInterface<DaffCartShippingRate> {
+  constructor(
+    private apollo: Apollo,
+    public shippingRateTransformer: DaffMagentoCartShippingRateTransformer
+  ) {}
+
+  list(cartId: string): Observable<DaffCartShippingRate[]> {
+    return this.apollo.query<ListShippingMethodsResponse>({
+      query: listShippingMethods,
+      variables: {cartId}
+    }).pipe(
+      map(result => result.data.cart.shipping_addresses[0].available_shipping_methods.map(item =>
```

Comment:
You could just do `map(result => result.data.cart.shipping_addresses[0].available_shipping_methods.map(this.shippingRateTransformer.transform))`

Reply:
That makes it more difficult to test what arguments were sent to the transformer since `map` will pass the index and array as 2nd and 3rg argument.